### PR TITLE
JSON-driven template generation

### PR DESCRIPTION
**Pull request by balanceiskey**
Addresses: https://github.com/equinusocio/vsc-material-theme/issues/13

Two notes:
* I have an additional commit coming that will address how we build icons. We'll move forward with utilizing `mustache` in place of `gulp-template` and `gulp-data` there as well. 
* I've swapped out most of the colors with what I think are the correct relevant keys from the variant files, but @equinusocio I could use your help in figuring out if they are correct. Additionally, I didn't swap in a replacement for `#252526`

### CODE REVIEW

**Comment by balanceiskey** [comment]
@equinusocio What's the best way to view an applied template in VSCode itself? What I mean is how does VSCode use say `Material-Theme-Darker.json` or `Material-Theme-Palenight.json`

**Comment by equinusocio** [comment]
Inside the package json under `contribute.themes` change the path to the json theme.

**Comment by equinusocio** [comment]
If you update from `develop` you will have the complete theme template, the `material-theme-darker.json` and the package.json updated

**Comment by skbolton** [comment]
I left a comment on the issue for this but I will bring it here. Are you guys thinking of moving the templating to gulp-template? I think it would be a good idea
I have some time today that I can put into this.

**Comment by balanceiskey** [comment]
@skbolton I didn't pursue `gulp-template` immediately because I was unfamiliar and was trying to get something going quickly, but I'm open to going that route if you think it'll be better. Can you clarify what the benefits would be?

**Comment by skbolton** [comment]
Nothing major it's just already in the project so one less dependency. And replacing the mustache call with template will do the same job. You would maybe have to change from {{}} to <%= > in the templates but even that might be configurable. Either way not a big deal I.

**Comment by equinusocio** [comment]
@balanceiskey All `#252526` should be replaced with `#212121`. Don't worry now, i have to polish all colors and colors names, so atm we should be sure that all work :)

**Comment by balanceiskey** [comment]
@skbolton Thanks for the feedback! I haven't had a chance in the past day or two, but was going to give converting the icons to `mustache` as well. You may be able to get to it first though, so let me know if you're interested!

**Comment by balanceiskey** [comment]
I had some time today @skbolton and @equinusocio so I took a stab at updating icons, have a look when you get a sec.

File: src/icons/icons-theme.json [hunk 1]
```diff
@@ -1,9 +1,8 @@
 {
-  //=include "partials/iconDefinitions.js"
-  //=include "partials/fileExtensions.js"
```
Comment (equinusocio) [inline src/icons/icons-theme.json]: `fileEstensions.js` is gone?

File: src/themes/theme-template.yml [hunk 2]
```diff
@@ -220,74 +220,74 @@ tokenColors:
...
       - source.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json
     settings:
-      foreground: '#FFCB6B'
+      foreground: '{{variant.scheme.base.yellow}}'
   - name: JSON Key - Level 2
     scope:
       - source.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json
     settings:
-      foreground: '#FFCB6B'
+      foreground: '{{variant.scheme.base.yellow}}'
```
Comment (equinusocio) [inline src/themes/theme-template.yml]: this is `{{variant.scheme.base.orange}}` if you want to fix it

File: src/icons/icons-theme.json (hunk 1 above)
Comment (balanceiskey) [inline src/icons/icons-theme.json]: Whoops 😬 I'll look into it and update

File: src/themes/theme-template.yml (hunk 2 above)
Comment (balanceiskey) [inline src/themes/theme-template.yml]: 👍

Comment (balanceiskey) [inline src/themes/theme-template.yml]: Was that strictly for this instance or for any use of '#FFCB6B'?

File: src/icons/icons-theme.json (hunk 1 above)
Comment (balanceiskey) [inline src/icons/icons-theme.json]: Should now be restored via https://github.com/equinusocio/vsc-material-theme/pull/15/commits/abdee55dc3d39af5bea9a22a9b30ef72aef28dd5

File: src/themes/theme-template.yml (hunk 2 above)
Comment (balanceiskey) [inline src/themes/theme-template.yml]: Resolved via: https://github.com/equinusocio/vsc-material-theme/pull/15/commits/4b037f8e5f8b8fab04513c7f77d44debc96c26e4

Comment (equinusocio) [inline src/themes/theme-template.yml]: > Was that strictly for this instance or for any use of #FFCB6B?
  Just for this instance.

**Comment by equinusocio** [comment]
I will merge to develop, now you can fetch and work on it. If you want to make fix/improvements open `feature/feature-name` branch or `fix/fix-name` branch from develop.

**Pull request merged**